File: 3. Longest Substring Without Repeating Characters/longest_substring_without_repeating_characters.rb
def longest_substring (text)
	len_text = text.length()
	longest_text = ""
	
	for a in 0...len_text
		for b in a...(len_text - a)
			substring = text[a, b]
			letter = text[a + b, 1]
			#puts (substring + "\t\t\t" + letter)
			
			if (substring.include?(letter))
				if (longest_text.length() < substring.length())
					longest_text = substring
				end
				break
			end
		end
	end
	
	return longest_text
end

w = longest_substring("pwwkew")
puts (w)
puts ("\n\n")

File: 3. Longest Substring Without Repeating Characters/longest_substring_without_repeating_characters.rs
fn longest_substring (text: String) -> String {
	let len_text: usize = text.len();
	let mut longest_text: String = "".to_string();
	
	for a in 0..len_text{
		for b in a..len_text{
			if (a + b) >= len_text{
				break;
			}
			
			let substring: String = text[a..(a + b)].to_string();
			let letter: &str = &text[(a + b)..(a + b + 1)];
			
			if substring.contains(letter){
				if longest_text.len() < substring.len(){
					longest_text = substring;
				}
				break;
			}
		}
	}
	return longest_text;
}

fn main () {
	let f: String = "pwwkew".to_string();
	let ret: String = longest_substring(f);
	
	println!("{}", ret);
}
